lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -498,6 +535,53 @@ def default_ami_for_region(region)
       end
     end
 
+    def create_winrm_transport(machine_spec, machine_options, instance)
+      remote_host = determine_remote_host(machine_spec, instance)
+      puts "remote host: #{remote_host}"
```

Comment:
I enjoy using the [pry](https://github.com/pry/pry) gem for debugging.  To use this with projects, I add
```ruby
gem "pry"
gem "pry-byebug"
gem "pry-stack_explorer"
```
to the Gemfile.  Then instead of your puts line above you can use `require 'pry'; binding.pry`.  This will open up an interactive pry session (similar to IRB) which allows you to inspect the current program state.  The 2 other gems also add step through and breakpoint functionality.  I find it quite robust!